sdv/tabular/base.py

```diff
@@ -471,6 +472,8 @@ def sample(self, num_rows=None, max_retries=100, max_rows_multiplier=10,
             else:
                 transformed_conditions_in_group = transformed_conditions.loc[condition_indices]
                 transformed_groups = transformed_conditions_in_group.groupby(transformed_columns)
+                if len(transformed_groups.groups) == 0:
```

Comment:
I think this implementation proposal breaks the possibility to fall back to conditioning based on reject-sampling if the constraints cannot transform the conditions data.

Instead of this, I suggest adding a new method that validates the input by iterating over the constraints and running the input data through their `is_valid` method if all the `constraint_columns` are contained in the `conditions`

Reply:
Just to clarify..  if there are no transformed groups, we should fall into the case on L461?

Reply:
In the end, we decided that the code should not reach the case of no transformed groups -- if the conditions are invalid, the transformed_conditions should have zero columns (a case that is already handled).